tests/unit/test_charm.py

```diff
@@ -83,5 +83,36 @@ def test_reporting_config(context, s3, all_worker, nginx_container, nginx_promet
         state_out = mgr.run()
 
         # THEN the worker should have the correct boolean value for reporting in analytics
-        config = get_worker_config_analytics(state_out.relations, "loki-cluster")
-        assert config == set_config
+        config = get_worker_config(state_out.relations, "loki-cluster", "analytics")
+        assert config["reporting_enabled"] == set_config
+
+@pytest.mark.parametrize(
+    "set_config, expected_config",
+    [
+        (0, {'retention_enabled': False, 'working_directory': '/loki/compactor'}),
+        (10, {'delete_request_store': 's3', 'retention_enabled': True, 'working_directory': '/loki/compactor'}),
+    ]
+)
+def test_retention(context, s3, all_worker, nginx_container, nginx_prometheus_exporter_container, set_config, expected_config):
+    """Ensure the coordinator sends the correct config for analytics and reporting to the worker."""
+    # GIVEN the config for reporting in Loki Coordinator is set to a boolean
+    config_value: str = set_config
+    config = {"retention-period": config_value}
+
+    state_in = State(
+        relations=[
+            s3,
+            all_worker,
+        ],
+        containers=[nginx_container, nginx_prometheus_exporter_container],
+        leader=True,
+        config=config
+    )
+
+    # WHEN a worker enters a relation to a coordinator
+    with context(context.on.relation_joined(all_worker), state_in) as mgr:
+        state_out = mgr.run()
+
+        # THEN the worker should have the correct boolean value for reporting in analytics
```

Comment:
Not sure why you mention "for reporting in analytics" here?
```suggestion
        # THEN the worker should correctly enable/disable retention
```